sql/moz-fx-data-marketing-prod/google_search_console/search_impressions_by_site/view.sql

```diff
@@ -1,36 +1,82 @@
 CREATE OR REPLACE VIEW
   `moz-fx-data-marketing-prod.google_search_console.search_impressions_by_site`
 AS
+WITH search_impressions_union AS (
+  SELECT
+    `date`,
+    site_url,
+    site_domain_name,
+    query,
+    FALSE AS is_anonymized,
+    search_type,
+    user_country_code,
+    device_type,
+    impressions,
+    clicks,
+    average_top_position
+  FROM
+    `moz-fx-data-marketing-prod.google_search_console_derived.search_impressions_by_site_v1`
+  WHERE
+    CASE
+      WHEN site_domain_name IN (
+          'addons.mozilla.org',
+          'blog.mozilla.org',
+          'getpocket.com',
+          'support.mozilla.org',
+          'www.mozilla.org'
+        )
+        THEN `date` < '2023-08-01'
+      WHEN site_domain_name = 'developer.mozilla.org'
+        THEN `date` < '2024-04-10'
+      ELSE FALSE
+    END
+  UNION ALL
+  SELECT
+    `date`,
+    site_url,
+    site_domain_name,
+    query,
+    is_anonymized,
+    search_type,
+    user_country_code,
+    device_type,
+    impressions,
+    clicks,
+    average_top_position
+  FROM
+    `moz-fx-data-marketing-prod.google_search_console_derived.search_impressions_by_site_v2`
+  WHERE
+    CASE
+      WHEN site_domain_name IN (
+          'addons.mozilla.org',
+          'blog.mozilla.org',
+          'getpocket.com',
+          'support.mozilla.org',
+          'www.mozilla.org'
+        )
+        THEN `date` >= '2023-08-01'
+      ELSE TRUE
+    END
+)
 SELECT
-  `date`,
-  site_url,
-  site_domain_name,
-  query,
-  FALSE AS is_anonymized,
-  search_type,
-  country_code,
-  device_type,
-  impressions,
-  clicks,
-  average_top_position
+  search_impressions.`date`,
+  search_impressions.site_url,
+  search_impressions.site_domain_name,
+  search_impressions.query,
+  mozfun.google_search_console.classify_site_query(
```

Comment:
Are Discover and Google News not relevant search types here so they don't need to be handled?

Reply:
Correct, Google only includes Discover and Google News in the URL-specific data, not the site-specific data.

Reply:
After further consideration, since Google could arbitrarily decide to include Discover and Google News in the site-specific data at some point, I'm going to change the ETLs to handle those the same way (probably by moving the logic into the `classify_site_query` function).